build/executorch-config.cmake

```diff
@@ -54,7 +54,12 @@ foreach(lib ${lib_list})
             If needed rebuild with the proper options in CMakeLists.txt")
     else()
         if("${lib}" STREQUAL "extension_module")
```

Comment:
Can we simplify this

if("${lib}" STREQUAL "extension_module" OR CMAKE_TOOLCHAIN_IOS)